Index: SQLAlchemy-0.8.0/test/dialect/test_postgresql.py
===================================================================
--- SQLAlchemy-0.8.0.orig/test/dialect/test_postgresql.py
+++ SQLAlchemy-0.8.0/test/dialect/test_postgresql.py
@@ -2836,8 +2836,8 @@ class HStoreTest(fixtures.TestBase):
         dialect = default.DefaultDialect()
         proc = self.test_table.c.hash.type._cached_bind_processor(dialect)
         eq_(
-            proc({"key1": "value1", "key2": "value2"}),
-            '"key2"=>"value2", "key1"=>"value1"'
+            ', '.join(sorted(proc({"key1": "value1", "key2": "value2"}).split(', '))),
+            '"key1"=>"value1", "key2"=>"value2"'
         )
 
     def test_parse_error(self):
@@ -2878,8 +2878,8 @@ class HStoreTest(fixtures.TestBase):
         dialect._has_native_hstore = False
         proc = self.test_table.c.hash.type._cached_bind_processor(dialect)
         eq_(
-            proc({"key1": "value1", "key2": "value2"}),
-            '"key2"=>"value2", "key1"=>"value1"'
+            ', '.join(sorted(proc({"key1": "value1", "key2": "value2"}).split(', '))),
+            '"key1"=>"value1", "key2"=>"value2"'
         )
 
     def test_result_deserialize_psycopg2(self):
