diff -up SQLAlchemy-0.8.1/test/orm/test_manytomany.py.bak SQLAlchemy-0.8.1/test/orm/test_manytomany.py
--- SQLAlchemy-0.8.1/test/orm/test_manytomany.py.bak	2013-04-29 18:05:36.721189243 -0700
+++ SQLAlchemy-0.8.1/test/orm/test_manytomany.py	2013-04-29 18:06:07.754995799 -0700
@@ -244,10 +244,10 @@ class M2MTest(fixtures.MappedTest):
         sess.add_all((t1, t2, t3, p1, p2, p3))
         sess.commit()
 
-        self.assert_result([t1],
+        self.assert_unordered_result([t1],
                     Transition, {'outputs':
                             (Place, [{'name': 'place3'}, {'name': 'place1'}])})
-        self.assert_result([p2],
+        self.assert_unordered_result([p2],
                         Place, {'inputs':
                                 (Transition, [{'name': 'transition1'},
                                                 {'name': 'transition2'}])})
